### PR TITLE
Add blog post 'Introducing Swift OpenAPI Generator'

### PR DESCRIPTION
**Pull request by czechboy0**
This PR is for an announcement blog post for Swift OpenAPI Generator.

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci test

**Comment by tomerd** [comment]
cc @cthielen @tkremenek

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+author: [honzadvorsky, simonjbeaumont]
+---
+
+We’re excited to announce a set of open source libraries designed to help both client and server developers streamline their workflow around HTTP communication using the industry‑standard OpenAPI specification.
+
+## What is OpenAPI?
+
+[OpenAPI](https://www.openapis.org/) is a specification for documenting HTTP services. An OpenAPI document is written in either YAML or JSON, and can be read by tools to help automate workflows, such as generating the necessary code to send and receive HTTP requests.
+
+By serving as the source of truth, OpenAPI solves the problem of communicating the API contract between a service and its clients. It removes the need to read potentially inaccurate, handwritten documentation, or to observe network traffic, just to figure out what the correct way to call a service is. It helps you avoid this time‑consuming, repetitive, and error‑prone work – not just when adopting a service for the first time, but also as the service continues to evolve.
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:15]: "just to figure out what the correct way to call a service is" => "just to figure out the correct way to call a service"

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        - message
+```
+
+This simple OpenAPI document defines the structure of the HTTP request and response, including the HTTP method, URL path and query parameters, HTTP status code and content type, and uses [JSON Schema](https://json-schema.org) to describe the structure of the response body.
+
+OpenAPI also allows defining additional details not shown above, such as query items and request bodies (see [OpenAPI 3.0.3 specification](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md) for more details).
+
+## Swift OpenAPI Generator
+
+[Swift OpenAPI Generator](https://github.com/apple/swift-openapi-generator) is a SwiftPM plugin that takes an OpenAPI document and generates either the client code to perform HTTP calls or the server code to handle those HTTP operations. The generated code translates between a type‑safe representation of each operation’s input and output and the underlying HTTP request and response.
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:71]: "either the client code to perform HTTP calls or the server code to handle those HTTP operations" => "either the client code to perform HTTP calls or the server code to handle those calls" ?

Comment (cthielen) [reply _posts/2023-06-12-introducing-swift-openapi-generator.md:71]: "each operation’s input and output and the underlying HTTP request" => "each operation’s input and output, and the underlying HTTP request" (comma after output)

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+We’re excited to announce a set of open source libraries designed to help both client and server developers streamline their workflow around HTTP communication using the industry‑standard OpenAPI specification.
+
+## What is OpenAPI?
+
+[OpenAPI](https://www.openapis.org/) is a specification for documenting HTTP services. An OpenAPI document is written in either YAML or JSON, and can be read by tools to help automate workflows, such as generating the necessary code to send and receive HTTP requests.
+
+By serving as the source of truth, OpenAPI solves the problem of communicating the API contract between a service and its clients. It removes the need to read potentially inaccurate, handwritten documentation, or to observe network traffic, just to figure out what the correct way to call a service is. It helps you avoid this time‑consuming, repetitive, and error‑prone work – not just when adopting a service for the first time, but also as the service continues to evolve.
+
+To see OpenAPI and its benefits in action, let’s consider a simple service called “GreetingService” that:
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:17]: Recommend GreetingService be in code formatting, not double quotes.

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+This simple OpenAPI document defines the structure of the HTTP request and response, including the HTTP method, URL path and query parameters, HTTP status code and content type, and uses [JSON Schema](https://json-schema.org) to describe the structure of the response body.
+
+OpenAPI also allows defining additional details not shown above, such as query items and request bodies (see [OpenAPI 3.0.3 specification](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md) for more details).
+
+## Swift OpenAPI Generator
+
+[Swift OpenAPI Generator](https://github.com/apple/swift-openapi-generator) is a SwiftPM plugin that takes an OpenAPI document and generates either the client code to perform HTTP calls or the server code to handle those HTTP operations. The generated code translates between a type‑safe representation of each operation’s input and output and the underlying HTTP request and response.
+
+So whether developing an app that serves as a client of “GreetingService” or implementing “GreetingService” itself, Swift OpenAPI Generator generates the networking ceremony code for you, allowing you to focus on the core logic.
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:73]: Recommend GreetingService be in code formatting, not double quotes.

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    * `openapi-generator-config.yaml`, a configuration file for the plugin, which controls whether to generate client or server code.
+
+>Tip: For more information about adopting the plugin in your Swift package or Xcode project, check out our step-by-step [tutorials](https://swiftpackageindex.com/apple/swift-openapi-generator/tutorials/swift-openapi-generator).
+
+### Using a Generated API Client
+
+When developing a client, such as an iOS app, you are provided with two generated types:
+
+* `APIProtocol`, a Swift protocol that contains one method per OpenAPI operation.
+    * In the case of “GreetingService”, it contains a single method called `getGreeting`.
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:95]: Recommend GreetingService be in code formatting, not double quotes.

File: _posts/2023-06-12-introducing-swift-openapi-generator.md (hunk 4 above)
Comment (cthielen) [reply _posts/2023-06-12-introducing-swift-openapi-generator.md:73]: Don't think "ceremony" is necessary to drive home the point that it's perfunctory code, no adjective should be fine.

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+## Swift OpenAPI Generator
+
+[Swift OpenAPI Generator](https://github.com/apple/swift-openapi-generator) is a SwiftPM plugin that takes an OpenAPI document and generates either the client code to perform HTTP calls or the server code to handle those HTTP operations. The generated code translates between a type‑safe representation of each operation’s input and output and the underlying HTTP request and response.
+
+So whether developing an app that serves as a client of “GreetingService” or implementing “GreetingService” itself, Swift OpenAPI Generator generates the networking ceremony code for you, allowing you to focus on the core logic.
+
+### Adopting the Plugin
+
+To use the Swift OpenAPI Generator plugin, you need to:
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:77]: Should this be a numbered list?

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+        // Print the greeting message.
+        print("👋 \(greeting.message)")
+    }
+case .undocumented(statusCode: let statusCode, _):
+    // Handle HTTP response status codes not documented in the OpenAPI document.
+    print("🥺 undocumented response: \(statusCode)")
+}
+```
+
+>Note: While the sample code above uses the [URLSession-based client transport](https://github.com/apple/swift-openapi-urlsession), there are other client transport implementations, and you can even create your own!
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:131]: "and you can even create your own!" => "or you can create your own."

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+let handler = Handler()
+
+// Register the generated routes on the transport.
+try handler.registerHandlers(on: transport)
+
+// Start the server.
+try app.run()
+```
+
+>Note: While the sample code above uses the [Vapor-based server transport](https://github.com/swift-server/swift-openapi-vapor), there are other server transport implementations, and you can even create your own!
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:175]: "and you can even create your own!" => "or you can create your own."

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+## Transport Implementations
+
+Swift OpenAPI Generator works with any HTTP client or server library by abstracting the HTTP library interface in protocols called [`ClientTransport`](https://swiftpackageindex.com/apple/swift-openapi-runtime/documentation/openapiruntime/clienttransport) and [`ServerTransport`](https://swiftpackageindex.com/apple/swift-openapi-runtime/documentation/openapiruntime/servertransport). Similar to projects like [swift-log](https://github.com/apple/swift-log#selecting-a-logging-backend-implementation-applications-only), Swift OpenAPI Generator uses the “API package” approach for greater extensibility.
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:179]: "in protocols called A and B" => "in the protocols A and B"

Comment (cthielen) [reply _posts/2023-06-12-introducing-swift-openapi-generator.md:179]: Recommend "API package" be in italics, not in double quotes.

File: _posts/2023-06-12-introducing-swift-openapi-generator.md [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+* [swift-server/swift-openapi-vapor](https://github.com/swift-server/swift-openapi-vapor)
+* [swift-server/swift-openapi-hummingbird](https://github.com/swift-server/swift-openapi-hummingbird)
+
+## What’s Next
+
+The project is being open sourced early in its development so the community can provide feedback and help us arrive at a stable 1.0 release.
+
+The initial focus was on implementing features defined by [version 3.0.3](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md) of the OpenAPI specification, with ongoing work toward supporting [OpenAPI 3.1](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md).
+
+While Swift OpenAPI Generator supports most of the commonly used features of OpenAPI, there are still several features left to implement, and we track our progress publicly using GitHub [issues](https://github.com/apple/swift-openapi-generator/issues).
```
Comment (cthielen) [inline _posts/2023-06-12-introducing-swift-openapi-generator.md:199]: "we track our progress publicly using" => "progress is tracked publicly using"

**Review by cthielen — COMMENTED** [review]
Looks good but please take a look at this last round of minor comments.

**Comment by alexandersandberg** [comment]
@swift-ci please test

**Comment by dempseyatgithub** [comment]
First, this is a great addition to the Swift ecosystem and a very informative well-written post, so thank you both for your work on it.

The last paragraph mentions providing feedback on the Swift forums, but there is no link to where in the forums.

Will OpenAPI Generator have its own sub-category in Related Projects?

It does not seem to have one yet, but once it does, the link to the forums should probably be added to the post.

**Comment by cthielen** [comment]
@swift-ci test

**Review by tomerd — APPROVED** [review]

**Pull request merged**